src/sentry/static/sentry/app/views/releasesV2/detail/overview/commitAuthorBreakdown.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import round from 'lodash/round';
+
+import {t, tn} from 'app/locale';
+import space from 'app/styles/space';
+import UserAvatar from 'app/components/avatar/userAvatar';
+import overflowEllipsis from 'app/styles/overflowEllipsis';
+import AsyncComponent from 'app/components/asyncComponent';
+import {percent} from 'app/utils';
+import {userDisplayName} from 'app/utils/formatters';
+import {Commit, User} from 'app/types';
+
+import {SectionHeading, Wrapper} from './styles';
+
+type Props = {
+  projectId: string;
+  orgId: string;
+  version: string;
+  commitCount: number;
+} & AsyncComponent['props'];
+
+type State = {
+  commits: Commit[];
+} & AsyncComponent['state'];
+
+class CommitAuthorBreakdown extends AsyncComponent<Props, State> {
+  getDefaultState() {
+    return {
+      ...super.getDefaultState(),
+    };
+  }
+
+  getEndpoints(): ReturnType<AsyncComponent['getEndpoints']> {
+    const {projectId, orgId, version} = this.props;
+
+    const commitsEndpoint = `/projects/${orgId}/${encodeURIComponent(
+      projectId
+    )}/releases/${encodeURIComponent(version)}/commits/`;
+
+    return [['commits', commitsEndpoint]];
+  }
+
+  getDisplayPercent(authorCommitCount: number): string {
+    const {commitCount} = this.props;
+
+    const calculatedPercent = round(percent(authorCommitCount, commitCount), 0);
+
+    return `${calculatedPercent < 1 ? '<1' : calculatedPercent}%`;
+  }
+
+  renderBody() {
+    // group commits by author
+    const groupedAuthorCommits = this.state.commits?.reduce(
+      (authorCommitsAccumulator, commit) => {
+        const email = commit.author?.email ?? 'unknown';
+
+        if (authorCommitsAccumulator.hasOwnProperty(email)) {
+          authorCommitsAccumulator[email].commitCount += 1;
+        } else {
+          authorCommitsAccumulator[email] = {
+            commitCount: 1,
+            author: commit.author,
+          };
+        }
+
+        return authorCommitsAccumulator;
+      },
+      {} as {[key: string]: {author: User | undefined; commitCount: number}}
```

Comment:
Should this be a named type?

Reply:
Sure I refactored it - I was thinking about it too, but it was only used there and didn't wrap to multiple lines so... well you know the rest